### ISSUE TITLE
Error using pipeline templates after erroneous template deployed until Orca restart

### ISSUE THREAD

**Issue by ctompkinson** [spinnaker/spinnaker#2038]
Error using pipeline templates after creating a pipeline using a template and selecting save

*Environment*
We are running Spinnaker in AWS using Halyard to deploy Spinnaker running version `1.4.2`

*Description*

When a pipeline template is added to spinnaker using `roer` and then a second correct pipeline template is added, when you attempt create a pipeline using that template from the UI and not select save, any future pipelines created from templates will throw an error.

*Steps to Reproduce*

1. Deploy a template containing Jinja templating that isn't satisfied using a variables using `roer pipeline-template publish template.yml`
2. Use `roer pipeline save` to create an instance of that template
3. Remove that pipeline using the UI
4. Remove the template using `roer pipeline-template delete`
5. Deploy a new template that provides all variables (or removes all templating)
6. Close the browser
7. Open spinnaker and attempt to create a new pipeline from a template (using the UI)

*Additional Details*

```
2017-10-16 16:35:52.759  INFO 26990 --- [0.1-8083-exec-6] c.n.s.orca.front50.Front50Service        : [chris.tompkinson@spaceapegames.co
2017-10-16 16:35:52.659  INFO 26990 --- [0.1-8083-exec-6] c.n.s.orca.front50.Front50Service        : [chris.tompkinson@spaceapegames.co
m] ---> HTTP GET http://localhost:8080/pipelineTemplates/BazTemplate
2017-10-16 16:35:52.759  INFO 26990 --- [0.1-8083-exec-6] c.n.s.orca.front50.Front50Service        : [chris.tompkinson@spaceapegames.co
m] <--- HTTP 200 http://localhost:8080/pipelineTemplates/BazTemplate (100ms)
2017-10-16 16:35:52.761  INFO 26990 --- [0.1-8083-exec-6] c.n.s.o.c.OperationsController           : [chris.tompkinson@spaceapegames.co
m] received pipeline ec856ed8-e259-4850-a64c-95455acfa2a9:{
  "keepWaitingPipelines" : false,
  "limitConcurrent" : true,
  "application" : "foo",
  "parameterConfig" : [ ],
  "name" : "abc",
  "stages" : [ {
    "storeType" : "ebs",
    "package" : "bar",
    "baseLabel" : "release",
    "type" : "bake",
    "extendedAttributes" : { },
    "enhancedNetworking" : true,
    "showAdvancedOptions" : false,
    "requisiteStageRefIds" : [ ],
    "baseOs" : "foo",
    "vmType" : "hvm",
    "name" : "Bake",
    "cloudProviderType" : "aws",
    "id" : "f0309276-ba18-4d60-aa36-2417f714d64f",
    "refId" : "bake",
    "user" : "bar"
  } ],
  "id" : "ec856ed8-e259-4850-a64c-95455acfa2a9",
  "notifications" : [ ],
  "errors" : [ {
    "severity" : "FATAL",
    "cause" : "Failed rendering stage",
    "message" : "failed rendering template expression"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: application, service, bakeRegions, applicationBaseOS, deploymentTags, deploymentStrategy",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: application, service, bakeRegions, applicationBaseOS, deploymentTags, deploymentStrategy",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: application, service, bakeRegions, applicationBaseOS, deploymentTags, deploymentStrategy",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: application, service, bakeRegions, applicationBaseOS, deploymentTags, deploymentStrategy",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Failed rendering stage",
    "message" : "failed rendering template expression"
  }, {
    "severity" : "FATAL",
    "cause" : "Failed rendering stage",
    "message" : "failed rendering template expression"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: service",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: service",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: service",
    "message" : "malformed template configuration"
  }, {
    "severity" : "FATAL",
    "cause" : "Missing variable values for: service",
    "message" : "malformed template configuration"
  } ],
  "mptRequestId" : "6bd13986-023a-41c3-b70d-e58eb44a239a",
  "trigger" : {
    "type" : "manual",
    "user" : "chris.tompkinson@spaceapegames.com"
  }
}
2017-10-16 16:35:52.761  INFO 26990 --- [0.1-8083-exec-6] c.n.s.o.c.OperationsController           : [chris.tompkinson@spaceapegames.com] not starting pipeline (plan: true): ec856ed8-e259-4850-a64c-95455acfa2a9
```

**Comment by SteveKMin** [comment]
This also happens to me when editing a template's configuration through the Spinnaker UI with an erroneous value, like inputting a string when expecting an int. 

In addition, it seems to break editing/creating any templated pipelines within the same spinnaker stack.

**Comment by jonsie** [comment]
This issue was resolved in this version of Orca:

https://github.com/spinnaker/orca/releases/tag/v3.21.7

The version of Orca that is published with 1.4.2 is https://github.com/spinnaker/orca/releases/tag/v3.21.2

**Comment by ctompkinson** [comment]
@jonsie Thanks for the update
